### PR TITLE
fixes #3887 fix(legacy): ensure pop percent are fixed before comparing

### LINKED ISSUE #3887: [Legacy] population percent extra changelogs with no changes
Extra changelogs are listed
https://experimenter.services.mozilla.com/experiments/measure-set-to-default-adoption-impact-of-changing-the-prompt-to-set-firefox-as-default-v2-win7/

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-546)
┆Issue Number: EXP-546

### PR DESCRIPTION
**Pull request by tiftran**
Because:

* division and precision

This commit:

* formats normandy percent decimal before comparing with population percent
* migration to prune empty population percent changelogs

### CODE REVIEW

File: app/experimenter/normandy/tasks.py [hunk 1]
```diff
@@ -165,11 +165,12 @@ def update_population_percent(experiment, recipe_data, filter_objects):
...
-            bucket_sample["count"] / bucket_sample["total"] * 100
-        )
-        if experiment.population_percent != percent_decimal:
-            changed_data = {"population_percent": percent_decimal}
+
+        percent_decimal = bucket_sample["count"] / bucket_sample["total"] * 100
+        rounded_decimal = format(percent_decimal, ".4f")
+
+        if str(experiment.population_percent) != rounded_decimal:
+            changed_data = {"population_percent": rounded_decimal}
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tasks.py:173]: Oh you're comparing them as strings, in theory that works, I was thinking of explicitly setting the Decimal precision and comparing them that way:

https://gist.github.com/jackiekazil/6201722#3-round-decimal-by-setting-precision

I can't think off the top of my head whether that makes a real difference, but if this goes sideways then we can try it that way.

File: app/experimenter/normandy/tests/test_tasks.py [hunk 2]
```diff
@@ -475,6 +475,43 @@ def test_firefox_version_no_updates_when_no_change(self):
 
         self.assertFalse(experiment.changes.filter(message="Added Version(s)").exists())
 
+    def test_live_experiment_rounds_population_percent_correctly(self):
```
Comment (jaredlockhart) [inline app/experimenter/normandy/tests/test_tasks.py:478]: I think we should have a test that replicates the case where unnecessary duplicates are created and asserts they aren't?

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Thnx @tiftran !

File: app/experimenter/normandy/tests/test_tasks.py (hunk 2 above)
Comment (tiftran) [reply app/experimenter/normandy/tests/test_tasks.py:478]: https://github.com/mozilla/experimenter/pull/3888/files#diff-2a8ff87f0439ee59f7dd865e60a59a2e0d52045a48fbcf807b7e4d532e0865f3R511
?

Comment (jaredlockhart) [reply app/experimenter/normandy/tests/test_tasks.py:478]: Oh oh! lol I'm dumb I thought that was an asserTrue yes good yep 👍

**Review by jaredlockhart — APPROVED** [review]
Yep is good thnx @tiftran ! 🎉

**Comment by jaredlockhart** [comment]
"In a minute there is time for division and precision which a PR can reverse" - The Bug Ticket of T Tran Prufrock by T.S. Eliot

**Pull request merged**